### ISSUE TITLE
Improve the quality of the code

### ISSUE THREAD

**Issue by sparkmuse** [sparkmuse/import-spark#1]
This is a learning project:

- Improve the quality of the code in general.
- Make it fully testable.

Help wanted:
- Make it faster :)

**Comment by jjjimenez100** [comment]
Let me help with this one. Any improvements in general, right? :)

**Comment by sarisxristos** [comment]
Can i help with it?

**Comment by sparkmuse** [comment]
Yes please. Feel free to help. One of the main problems is speed :)
